js/load.js

```diff
@@ -99,18 +100,116 @@ import ListPanel from '@/components/ListPanel/ListPanel.vue';
 import VueRegistry from './classes/VueRegistry.js';
 import i18nPlugin from '@/piniaPlugins/i18n.plugin.js';
 
+// Register global components
+VueRegistry.registerComponent('Badge', Badge);
```

Comment:
If all components are properly named, I think we could extract the name dynamically from them (e.g. `Vue.component(component.name, component);`), and move these registrations into a loop.

I don't know what's the reasoning for registering `Spinner` and `PkpSpinner` (I imagine that it started with `Button`, then conflicted with real HTML elements, and was moved to `PkpButton`). Perhaps it's a good time to standardize the naming, given that probably almost nobody extended the Vue components 🤔

Reply:
Naming was bit mixed, likely due some name conflicts with native elements.  So when I was exposing bigger part of ui-library to be available for plugins for 3.4, I kept the existing ones to avoid any surprises as 3.4 was already released and basically just exposed everything with pkp prefix.  But its good point for 3.5 that might be good opportunity to get rid of the ones without prefix.  Will consider it in separate PR.

Getting name from the component itself is also good suggestion, will keep it in mind when I review this logic. I need to consider whether to possibly expose all components globally, so they can be easily replaced from plugin via registry, that could be quite powerful way if someone needs some more significant customisation.